transcom/mymove
# INT B-18314-adding-multi-move-data-to-dashboard

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3a873920)

## Summary

As we transition to multiple moves and integrating the new multiple move landing page, we need to stop using test data and use the user's actual data returned from the recently implemented [`getAllMoves` API endpoint ](https://github.com/transcom/mymove/pull/11866). 

This PR does the following:
- adds a small enhancement to return the move status & origin address from the backend
- removed dummy test data being used
- implemented redux into `MultiMovesLandingPage` and added appropriate wrappers
- using `serviceMember` data to populate first name & last name
- updated helper text box that is based on if the user has previous moves or not
- updated null value handling when generating components (async issues were causing some errors)
- updated values used in components to match what is being sent from API endpoint
- updated `onboarding.js` to add `serviceMemberMoves` to state on load
- updated all relevant tests

### How to test

1. Access MM as a customer
2. Log in and you should see the multi-move dashboard
3. The data displayed to you should be accurate and relevant to the user that you are signed in as
4. Expand the move container to show shipments (if any) - if not, you should see "no shipments in move yet"

![Screenshot 2024-02-02 at 9 04 46 AM](https://github.com/transcom/mymove/assets/136510600/93478050-4ef0-4b8e-8638-9da130d9faf1)
